### PR TITLE
MySQL 8.0 should be started with mysqlx=0

### PR DESCRIPTION
**Pull request by deepthi**
Signed-off-by: deepthi <deepthi@planetscale.com>

@derekperkins are there other things we need to fix up for 8.0?

### CODE REVIEW

File: config/mycnf/master_mysql80.cnf [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+innodb_use_native_aio = 0
+
+# Semi-sync replication is required for automated unplanned failover
+# (when the master goes away). Here we just load the plugin so it's
+# available if desired, but it's disabled at startup.
+#
+# If the -enable_semi_sync flag is used, VTTablet will enable semi-sync
+# at the proper time when replication is set up, or when masters are
+# promoted or demoted.
+plugin-load = rpl_semi_sync_master=semisync_master.so;rpl_semi_sync_slave=semisync_slave.so
```
Comment (deepthi) [inline config/mycnf/master_mysql80.cnf]: is this something we can delete for 8.0?

**Comment by derekperkins** [comment]
There are definitely more things to change. I didn't know we had decided to add a new flavor. Why are you setting `mysqlx=0`? These are two that are definitely required. 

```
default_authentication_plugin   = mysql_native_password
secure_file_priv                = NULL
```

Offhand, other ones that we might do are 
```
character-set-server            = utf8mb4
collation-server                = utf8mb4_0900_ai_ci
```

@sjmudd is probably a better judge of good defaults.

**Comment by deepthi** [comment]
> There are definitely more things to change. I didn't know we had decided to add a new flavor. Why are you setting `mysqlx=0`? These are two that are definitely required.
> 
> ```
> default_authentication_plugin   = mysql_native_password
> secure_file_priv                = NULL
> ```
> Offhand, other ones that we might do are
> 
> ```
> character-set-server            = utf8mb4
> collation-server                = utf8mb4_0900_ai_ci
> ```
> @sjmudd is probably a better judge of good defaults.

re: mysqlx, either we need to disable it or change our tests to provide a separate port to each mysql instance. otherwise all the test mysqlds try to use the same default port for mysqlx and fail to start.

File: config/mycnf/master_mysql80.cnf (hunk 1 above)
Comment (dkhenry) [reply config/mycnf/master_mysql80.cnf]: according to the manual you still need the plugin in mysql 8.0 

https://dev.mysql.com/doc/refman/8.0/en/replication-semisync-installation.html

**Comment by sougou** [comment]
@derekperkins: I'll let you review and merge this.

**Comment by derekperkins** [comment]
The cnf file looks fine. I'm not super familiar with where new flavors need to be registered, but comparing with @dkhenry adding a MariaDB 10.3 flavor, there are a couple places that seem to be missing. https://github.com/vitessio/vitess/pull/4193/files

https://github.com/vitessio/vitess/blob/master/go/vt/vttest/environment.go#L120
https://github.com/vitessio/vitess/blob/master/py/vttest/mysql_flavor.py#L99

**Comment by deepthi** [comment]
@derekperkins thanks for the pointers. I fixed those and now I can get docker tests passing locally with the mysql80 flavor.

**Pull request merged**